ember_debug/render-debug.js

```diff
@@ -35,9 +36,37 @@ export default EmberObject.extend(PortMixin, {
   },
 
   sendAdded(profiles) {
+    if (this.shouldHighlightRender) {
+      profiles.forEach(profile => {
+        this._hightLightNode(profile);
+      })
+    }
     this.sendMessage('profilesAdded', { profiles });
   },
 
+  _hightLightNode({ viewGuid, children }) {
+    const hasChildren = children?.length > 0;
+    if (viewGuid) {
+      this._createOutline(viewGuid, hasChildren)
+    }
+    if (hasChildren) {
+      children.forEach(childNode => {
+        this._hightLightNode(childNode);
+      })
+    }
+  },
+
+  _createOutline(viewGuid, hasChildren) {
+    const element = document.getElementById(viewGuid);
```

Comment:
as I understand it wll work only with ember components with element property?

Reply:
Good question and thank you for the feedback. 
It should work for all the ember component, which should have `id="ember101"`. We do not need highlight the other dom which is within the ember component.
However, can you tell me in which case ember component does not have element property?

Reply:
it's about glimmer component itself, or ember component with element = '';

we could get roots of glimmer component using renderTree api (already used in ember-inspector), example of how we could find roots of glimmer component <https://github.com/microsoft/playwright/pull/8444/files#diff-758a9d9fc7482116b104ad9be2c48f82e39f039e600d571afb3c7f65f6ae213bR142>

Reply:
I looked into the renderTree API. From what I understand, it can match the node and highlight it.
However, render-debug use the Ember Instrumentation API. https://api.emberjs.com/ember/2.14/classes/Ember.Instrumentation
https://github.com/emberjs/ember-inspector/blob/master/ember_debug/render-debug.js#L74
Instrumentation provides a payload for the component which got rendered. The payload does not have the exact dom.
In the other word, we do not get the newly rendered node directly from the browser. I take a screenshot of the console log of the payload
![Screen Shot 2021-09-18 at 12 10 21 AM](https://user-images.githubusercontent.com/20895144/133879870-5ed2cb88-a8c4-4a3d-9c50-39b51c528d1a.png)
I do see the "element" key you are talking about. Do you think we can match the component in the renderTree using the payload? Does Ember has any util can convert to node?

Reply:
nice to meeting. I changed the highlight logic as you suggested.